lane.py

```diff
@@ -767,16 +779,40 @@ def scoring_functions(self, filter):
         return filter.featurability_scoring_functions(self.random_seed)
 
 
-class SearchFacets(FacetsWithEntryPoint):
+class SearchFacets(Facets):
     """A Facets object designed to filter search results.
 
     Most search result filtering is handled by WorkList, but this
     allows someone to, e.g., search a multi-lingual WorkList in their
     preferred language.
     """
 
-    def __init__(self, entrypoint=None, media=None, languages=None, **kwargs):
-        super(SearchFacets, self).__init__(entrypoint, **kwargs)
```

Comment:
The big change I had to make is changing the `SearchFacets` constructor so that it's compatible with the `Facets` constructor. Previously you could just pass in an EntryPoint; now the first argument is supposed to be a Library.

I don't think this is going to be a huge deal. Outside of tests, the only time we should create a SearchFacets object is in `facets_from_request`.

Reply:
But now it's just `**kwargs` that get passed for `**constructor_kwargs`, right?

Reply:
Yes, we add library, collection, availability, and order to the kwargs (all of which are required by the Facets constructor) if there is no current value, and then pass them on into the Facets constructor.